core/domain/activity_domain.py

```diff
@@ -28,17 +28,38 @@ class ActivityReference(object):
     """
 
     def __init__(self, activity_type, activity_id):
+        """Constructs an ActivityReference instance.
+
+        Args:
+          activity_type: str.
+          activity_id:   str.
+        """
         self.type = activity_type
         self.id = activity_id
 
     def get_hash(self):
+        """Gets instance hash.
+
+        Returns:
+          A str in the form of 'type:id'.
+        """
         return '%s:%s' % (self.type, self.id)
 
     def validate(self):
+        """Checks if instance type is valid.
+
+        Raises:
+          Exception: type is invalid.
```

Comment:
Indentation. Also, capitalize first word after the colon.

Also: "type is invalid" slightly lacks detail. How about

    Exception: The activity type is invalid.